### PR TITLE
dh-310 investment document endpoints

### PR DESCRIPTION
**Pull request by jim68000**
Also added actual_land_date, which was missing

### CODE REVIEW

**Comment by jim68000** [comment]
Sorry had to make this much more complicated in order to deal with the requirement that you can have multiple documents per field. 

The assumption is that a> it's the node.js layer that actually posts the documents to S3, and simply tells the back end the location of the file.

Therefore we need two new tables to deal with this: a documentable_fields metadata table that gives ids of the fields within InvestmentProject that can take documents - eg, actual_land_date_documents etc. This will allow us to also have a 'general' magic id for documents that aren't related to a field, should we need it.

id | name
---|----
abcd...|foreign_investment_documents

Then we need a table which links documents to projectids and the field they belong to. So, id, foreign key for project, foreign key for field (we could also do this with enaums, but that's less flexible), uri of document

id | fk_project | fk_field | uri
---|---------|--------|----
abc... | def... | ghi... | https://example.com/thing/thing/Document.xlsx

File: generator/specs/paths/investment.yml [hunk 1]
```diff
@@ -329,19 +329,93 @@
...
+        type: string
+        format: uuid
+      - in: body
+        name: document
+        description: object that contains the document URL and which the id of which section it belongs to
+        required: true
+        schema:
+          type: object
+          properties:
+            documentUri:
```
Comment (reupen) [inline generator/specs/paths/investment.yml]: So far we've been using lower case with underscores for property names, so `document_uri` (or just `uri`)?

File: generator/specs/paths/investment.yml [hunk 2]
```diff
@@ -329,19 +329,93 @@
...
+          description: document added
+        400:
+          description: input invalid
+        404:
+          description: investment is already archived or does not exist
+  /v3/investment/document/{documentId}:
+    delete:
+      tags:
+      - investment
+      summary: Tells the back end that a document has successfully uploaded and where to find it
```
Comment (reupen) [inline generator/specs/paths/investment.yml]: Summary needs changing here.

File: generator/specs/paths/investment.yml (hunk 1 above)
Comment (jim68000) [inline generator/specs/paths/investment.yml]: I like uri

File: generator/specs/paths/investment.yml (hunk 2 above)
Comment (jim68000) [inline generator/specs/paths/investment.yml]: OK, done

**Pull request merged**